src/FNA3D_Driver_Vulkan.c

```diff
@@ -2372,9 +2372,12 @@ static VulkanBuffer* CreateBuffer(
 	{
 		usageFlags = VK_BUFFER_USAGE_TRANSFER_SRC_BIT;
 	}
-	else if (resourceAccessType == RESOURCE_ACCESS_TRANSFER_WRITE)
+
+	if (	resourceAccessType == RESOURCE_ACCESS_TRANSFER_READ ||
+		resourceAccessType == RESOURCE_ACCESS_TRANSFER_WRITE	)
 	{
-		usageFlags = VK_BUFFER_USAGE_TRANSFER_DST_BIT;
+		/* FIXME: There's probably a better way to handle this... -caleb */
+		usageFlags |= VK_BUFFER_USAGE_TRANSFER_DST_BIT;
```

Comment:
we probably want VK_BUFFER_USAGE_TRANSFER_SRC_BIT for RESOURCE_ACCESS_TRANSFER_READ, yeah?

Reply:
The reason I or'd the dst bit even for RESOURCE_ACCESS_TRANSFER_READ is because staging buffers are created with that access, and we need them to be both readable and writable. I guess we could add a RESOURCE_ACCESS_TRANSFER_READ_WRITE?